30_regexp_lab/relab.txt

```diff
@@ -1,17 +1,25 @@
 Use grep to print only the lines from "input" for which:
 
 1. There is a binary number in the line
+	grep -E -w '[10]+' input.txt | grep -E -v '[10]+[),]'
 
 2. There is an even binary number in the line
-
+	grep -E -w '[10]*0' input.txt | grep -E -v '[10]*0[),]'
+	
 3. Line includes a hexadecimal number
+	grep -E -w -i '0*x*[1-9a-f]+' input.txt | grep -v '^1'
```

Comment:
This regex will match ```00000xxxxxxxxxxff``` and it will miss ```0xff0f`` or any hex number with a 0...  (note that 0 is a valid hex digit)

If you want to match hex numbers with hex prefix (0x), I recommend using the same  trick that we used for binary, just adding "0x" at the beginning:``` grep -E '(^|\s)0x[0-9,a-f]+($|\s)' ```

Also note that this pattern will not match all hex numbers in a line, because once it matches a pattern, it "eats" the following white-space.. but we don't really care, we only want to count lines :)